make/testing_flags.mk

```diff
@@ -1,6 +1,46 @@
-ITEST_FLAGS = 
+include make/compile_flags.mk
+
+ITEST_FLAGS =
+TEST_FLAGS =
+DEV_TAGS = dev
 
 # Define the integration test.run filter if the icase argument was provided.
 ifneq ($(icase),)
 ITEST_FLAGS += -test.run="TestLightningTerminal/$(icase)"
 endif
+
+# If a specific unit test case is being targeted, construct test.run filter.
+ifneq ($(case),)
+TEST_FLAGS += -test.run=$(case)
+UNIT_TARGETED = yes
+endif
+
+# If specific package is being unit tested, construct the full name of the
+# subpackage.
+ifneq ($(pkg),)
+UNITPKG := $(PKG)/$(pkg)
+COVER_PKG := $(PKG)/$(pkg)
+UNIT_TARGETED = yes
+GOLIST = echo '$(PKG)/$(pkg)'
+endif
+
+# Add any additional tags that are passed in to make.
+ifneq ($(tags),)
+DEV_TAGS += ${tags}
+endif
+
+# UNIT_TARGETED is undefined iff a specific package and/or unit test case is
+# not being targeted.
+UNIT_TARGETED ?= no
+
+# If a specific package/test case was requested, run the unit test for the
+# targeted case. Otherwise, default to running all tests.
+ifeq ($(UNIT_TARGETED), yes)
+UNIT := $(GOTEST) -tags="$(DEV_TAGS) $(COMPILE_TAGS)" $(TEST_FLAGS) $(UNITPKG)
```

Comment:
`make unit case=TestObfuscateConfig` would not run, but only adding the package will add the test files in order for the test to be discoverable: `make unit case=TestObfuscateConfig pkg=firewall`. Do we want that, because it seems like an extra step that we could avoid?

Reply:
note that this was mentioned in this commit message: https://github.com/lightninglabs/lightning-terminal/pull/928/commits/824a3c259c84fbde77e24febcf6c17e0fc5c9253

Reply:
you have to specify package. Otherwise you have to search through all packages. Im sure there is a helper we can add later on but think this is fine for now?

Reply:
> note that this was mentioned in this commit message: https://github.com/lightninglabs/lightning-terminal/commit/824a3c259c84fbde77e24febcf6c17e0fc5c9253

ah yes, now that makes sense! I just noticed that in lnd it's the same (usually I'm running tests differently), so it makes sense to keep things consistent across projects. it really takes a lot longer if we'd include all packages with 24s vs 2s :+1: